qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/datamodel/ExtendedDataModelTemplate.java

```diff
@@ -16,21 +16,35 @@
 import java.util.stream.Collectors;
 
 import com.redhat.qute.commons.QuteJavaDefinitionParams;
+import com.redhat.qute.commons.datamodel.DataModelFragment;
 import com.redhat.qute.commons.datamodel.DataModelParameter;
 import com.redhat.qute.commons.datamodel.DataModelTemplate;
 
-public class ExtendedDataModelTemplate extends DataModelTemplate<ExtendedDataModelParameter> {
+public class ExtendedDataModelTemplate extends DataModelTemplate<ExtendedDataModelParameter>
+		implements DataModelSourceProvider {
 
 	public ExtendedDataModelTemplate(DataModelTemplate<DataModelParameter> template) {
 		super.setTemplateUri(template.getTemplateUri());
 		super.setSourceType(template.getSourceType());
 		super.setSourceMethod(template.getSourceMethod());
 		super.setSourceField(template.getSourceField());
 		super.setParameters(createParameters(template.getParameters(), this));
+		super.setFragments(createFragments(template.getFragments(), this));
 	}
 
-	private List<ExtendedDataModelParameter> createParameters(List<DataModelParameter> parameters,
+	private List<DataModelFragment<ExtendedDataModelParameter>> createFragments(
+			List<DataModelFragment<DataModelParameter>> fragments,
 			ExtendedDataModelTemplate template) {
+		if (fragments == null) {
+			return Collections.emptyList();
+		}
+		return fragments.stream() //
+				.map(fragment -> new ExtendedDataModelFragment(fragment)) //
+				.collect(Collectors.toList());
+	}
+
+	static List<ExtendedDataModelParameter> createParameters(List<DataModelParameter> parameters,

```

Comment:
Any reason this is package private instead of private/protected?

Reply:
fixed